src/engine/GroupBy.cpp

```diff
@@ -447,6 +450,52 @@ bool GroupBy::computeGroupByForSingleIndexScan(IdTable* result) {
   return true;
 }
 
+// ____________________________________________________________________________
+bool GroupBy::computeGroupByObjectWithCount(IdTable* result) {
+  // The child must be an `IndexScan` with exactly two variables.
+  auto* indexScan =
+      dynamic_cast<const IndexScan*>(_subtree->getRootOperation().get());
+  if (!indexScan || indexScan->numVariables() != 2) {
+    return false;
+  }
+  const auto& permutedTriple = indexScan->getPermutedTriple();
+  const auto& vocabulary = getExecutionContext()->getIndex().getVocab();
+  std::optional<Id> col0Id = permutedTriple[0]->toValueId(vocabulary);
+  if (!col0Id.has_value()) {
+    return false;
+  }
+
+  // There must be exactly one GROUP BY variable and the result of the index
+  // scan must be sorted by it.
+  if (_groupByVariables.size() != 1) {
+    return false;
+  }
+  const auto& groupByVariable = _groupByVariables.at(0);
+  AD_CORRECTNESS_CHECK(
+      *(permutedTriple[1]) == groupByVariable,
+      "Result of index scan for GROUP BY must be sorted by the "
+      "GROUP BY variable, this is a bug in the query planner",
+      permutedTriple[1]->toString(), groupByVariable.name());
+
+  // There must be exactly one alias, which is a non-distinct count of one of
+  // the two variables of the index scan.
+  auto countedVariable = getVariableForNonDistinctCountOfSingleAlias();
+  bool countedVariableIsOneOfIndexScanVariables =
+      countedVariable.has_value() &&
+      (countedVariable.value() == *(permutedTriple[1]) ||
+       countedVariable.value() == *(permutedTriple[2]));
```

Comment:
```suggestion
  bool countedVariableIsOneOfIndexScanVariables =
      (countedVariable.value() == *(permutedTriple[1]) ||
       countedVariable.value() == *(permutedTriple[2]));
```
This should work directly with `optional`s.